File: UnitTest.cpp
#include <iostream>
#include "Solution.hpp"

struct testcase {
    vector<string> words;
    vector<vector<int>> queries;
    vector<int> outputs;
};

class UnitTest {
private:
    vector<testcase> testcases;
    Solution obj;
public:
    UnitTest() {
        testcases = {
            {{"aba","bcb","ece","aa","e"}, {{0,2},{0,4},{1,1}}, {2,3,0}},
            {{"a","e","i"}, {{0,2},{0,1},{2,2}}, {3,2,1}}
        };
    }

    void test() {
        for(int i = 0; i < testcases.size(); ++i) {
            vector<string> words = testcases[i].words;
            vector<vector<int>> queries = testcases[i].queries;
            vector<int> outputs = testcases[i].outputs;

            vector<int> results = obj.vowelStrings(words, queries);
            bool pass = true;
            if(results.size() == outputs.size()) {
            for(int j = 0; j < results.size(); ++j)
                if(results[i] != outputs[i]) { pass = false; break; }
            }
            else pass = false;

            cout << "TestCase " << i+1 << ": " << (pass ? "passed" : "failed") << endl;
        }
    }
};

int main() {
    UnitTest test;
    test.test();
}

File: UnitTest.py
from Solution import Solution
from timeout_decorator import timeout
import unittest

class UnitTest(unittest.TestCase):
    def setUp(self):
        self.__testcases = {1: (["aba","bcb","ece","aa","e"], [[0,2],[1,4],[1,1]], [2,3,0]),
                            2: (["a","e","i"], [[0,2],[0,1],[2,2]], [3,2,1])}
        self.__obj = Solution()
        return super().setUp()
    
    @timeout(0.5)
    def test_case_1(self):
        words, queries, outputs = self.__testcases[1]
        results = self.__obj.vowelStrings(words = words, queries = queries)
        self.assertIsInstance(results, list)
        self.assertTrue(all(r == o for r, o in zip(results, outputs)))

    @timeout(0.5)
    def test_case_2(self):
        words, queries, outputs = self.__testcases[2]
        results = self.__obj.vowelStrings(words = words, queries = queries)
        self.assertIsInstance(results, list)
        self.assertTrue(all(r == o for r, o in zip(results, outputs)))

if __name__ == '__main__': unittest.main()